src/ValidationResult.php

```diff
@@ -30,4 +30,25 @@ public function getViolations(): array {
 		return $this->violations;
 	}
 
+	/**
+	 * Set source of an error in all violations.
+	 *
+	 * This method is meant for validation results that come from one source,
+	 * e.g. a validator like {@see WMDE\FunValidators\Validators\RequiredFieldValidator}
+	 * or {@see WMDE\FunValidators\Validators\StringLengthValidator}
+	 *
+	 * DO NOT call it on validation results that come from validators
+	 * that validate several sources. In those cases, the responsibility to
+	 * set the source is with the validator and not its calling code.
+	 */
+	public function setSource( string $sourceName ): self {
```

Comment:
I was wondering (as this is a special kind of "getter"), whether we could also give it a more "special" sounding name, like
`setSourceForAllViolations`
`setSourceForExistingViolations`

Reply:
Excellent idea, that'll make it also more clear what the method does on the call site. Changing the public API is a big endeavour that I would shy away from, but since we're doing a BC break anyway, we can do this easily. I've deprecated `setSource` for the dot release (internally it calls `setSourceForAllViolaitions`) and will remove it in the next major release that removes `CanValidateField`